AppCenterDistribute/AppCenterDistribute/MSDistribute.m

```diff
@@ -807,6 +809,31 @@ - (NSMutableDictionary *)getReportingParametersForUpdatedRelease:(NSString *)upd
   return reportingParameters;
 }
 
+- (void)changeDistributionGroupIdAfterAppUpdateIfNeeded:(NSString *)currentInstalledReleaseHash {
+  NSString *lastDownloadedDistributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDownloadedDistributionGroupIdKey];
+  if (lastDownloadedDistributionGroupId == nil) {
+    return;
+  }
+
+  // Skip if the current release was not updated.
+  NSString *lastDownloadedReleaseHashes = [MS_USER_DEFAULTS objectForKey:kMSDownloadedReleaseHashKey];
+  if ((lastDownloadedReleaseHashes == nil) || ([lastDownloadedReleaseHashes rangeOfString:currentInstalledReleaseHash].location == NSNotFound)) {
+    return;
+  }
+
+  // Skip if the group ID of an updated release is the same as the current one.
+  NSString *currentDistributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDistributionGroupIdKey];
+  if ((currentDistributionGroupId == nil) || ([lastDownloadedDistributionGroupId isEqualToString:currentDistributionGroupId] == NO)) {
+
+    // Set group ID from downloaded release details if an updated release was downloaded from another distribution group.
+    MSLogDebug([MSDistribute logTag], @"Current group ID doesn't match the group ID of downloaded release, updating current group id: %@", lastDownloadedDistributionGroupId);
+    [MS_USER_DEFAULTS setObject:lastDownloadedDistributionGroupId forKey:kMSDistributionGroupIdKey];
+  }
+
+  // Remove saved downloaded group ID.
+  [MS_USER_DEFAULTS removeObjectForKey:kMSDownloadedDistributionGroupIdKey];
```

Comment:
Sounds like `kMSDownloadedDistributionGroupIdKey` will be used in a same session. Why don't you pass this value as a parameter?

Reply:
It will be used after restart of an app. If a downloaded release was actually installed then SDK will use group ID of downloaded release to update stored group ID.

SDK checks if the hash of currently running app matches the hash of the release that was recently downloaded. This logic is executed only if they match. If hashes match then it means that downloaded release was installed and we need to check the group ID. If the user downloaded a release from another group ID then we need to update stored group ID.

Please let me know if you have any questions.